Chimera Ortho refactor

noroadsleft opened the pull request:
## Commit Log

### Layout macro refactor (8ab1cfe)

- Renamed KC_KEYMAP to LAYOUT_kc
- Renamed KEYMAP to LAYOUT
- moved LAYOUT above LAYOUT_kc (Hoping this encourages users to use LAYOUT.)
- updated layout macro name in info.json
- white space changes for readability

### Keymap refactor: default (1f3d81d)

Default keymap now uses `#include QMK_KEYBOARD_H` and LAYOUT macro (from KC_KEYMAP/LAYOUT_kc). Keymap was also refactored to use process_record_user function instead of deprecated action_get_macro from TMK.

Also moved LONGPRESS_DELAY and LAYER_TOGGLE_DELAY definitions to config.h.

### Keymap refactor: dcompact (2922200)

- Changed chimera_ortho.h include for QMK_KEYBOARD_H
- Updated layout name on keymap layers

noroadsleft commented:
This is what I get for trying to clear out some long-overdue changes; I forgot I never figured out exactly why the `gordon` keymap doesn't compile.

I know it's a Tap Dance thing, but as I've never used Tap Dance I don't quite know how to debug this.

drashna commented on keyboards/chimera_ortho/keymaps/default/keymap.c:
```diff
@@ -1,20 +1,20 @@
 // The underscores don't mean anything - you can have a layer called STUFF or any other name.
 // Layer names don't all need to be of the same length, obviously, and you can also skip them
 // entirely and just use numbers.
 enum chimera_ortho_layers
```
Could this be changed so its:
```c
enum chimera_ortho_layers {
```

noroadsleft commented:
Interesting:

I decided to locally checkout [the merge commit](https://github.com/qmk/qmk_firmware/tree/383a3c1e08d0f4e00b1c6f07e8468c5eac544eb4) when the `gordon` keymap was added.

It doesn't compile there, either.

noroadsleft commented:
## New Commits

### Keymap refactor: gordon (cd836db)

- Changed chimera_ortho.h include for QMK_KEYBOARD_H
- Removed redundant includes of action_layer.h and process_tap_dance.h (superseded by QMK_KEYBOARD_H)
- Removed redundant definitions for KC_NO and KC_TRNS
- Updated layout name on keymap layers

----

At this point, I'm confident I haven't broken anything any worse than it was already broken.

@drashna If you're good with the current state, I'm calling this done.

drashna commented:
Default keymap compiles, and the rest of the errors are from a keymap that needs updating (and has already been tagged). 

So, thanks!
